cargo-screeps/src/build.rs

```diff
@@ -188,65 +175,105 @@ if( typeof Rust === "undefined" ) {
         "^{}",
         make_into_slightly_less_brittle_regex(expected_prefix)
     ))?;
 
+    let expected_suffix = regex::Regex::new(&format!(
+        "{}$",
+        make_into_slightly_less_brittle_regex(expected_suffix)
+    ))?;
+
     debug!("expected prefix:\n```{}```", expected_prefix);
+    debug!("expected suffix:\n```{}```", expected_suffix);
 
     let prefix_match = expected_prefix.find(input).ok_or_else(|| {
         format_err!(
             "'cargo web' generated unexpected JS prefix! This means it's updated without \
-             'cargo screeps' also having updates. Please report this issue to \
+             'cargo screeps' also having updated. Please report this issue to \
              https://github.com/daboross/screeps-in-rust-via-wasm/issues and include \
              the first ~30 lines of {}",
             file_name.display(),
         )
     })?;
 
-    let initialize_function = &input[prefix_match.end()..];
+    let suffix_match = expected_suffix.find(input).ok_or_else(|| {
+        format_err!(
+            "'cargo web' generated unexpected JS suffix! This means it's updated without \
+             'cargo screeps' also having updated. Please report this issue to \
+             https://github.com/daboross/screeps-in-rust-via-wasm/issues and include \
+             the last ~30 lines of {}",
+            file_name.display(),
+        )
+    })?;
+
+    let initialize_function = &input[prefix_match.end()..suffix_match.start()];
 
     // screeps doesn't have `console.error`, so we define our own `console_error` function,
     // and call it.
     let initialize_function = initialize_function.replace("console.error", "console_error");
 
-    let wasm_module_name = wasm_filename
+    let wasm_module_name = config
+        .output_wasm_file
         .file_stem()
         .ok_or_else(|| {
             format_err!(
                 "expected output_wasm_file ending in a filename, but found {}",
-                wasm_filename.display()
+                config.output_wasm_file.display()
             )
-        })?.to_str()
+        })?
+        .to_str()
         .ok_or_else(|| {
             format_err!(
                 "expected output_wasm_file with UTF8 filename, but found {}",
-                wasm_filename.display()
+                config.output_wasm_file.display()
             )
         })?;
 
+    let initialization_header: Cow<'static, str> = match config.initialization_header_file.as_ref()
+    {
+        Some(header_file) => fs::read_to_string(header_file)?.into(),
+        None => include_str!("../resources/default_initialization_header.js").into(),
+    };
+
     Ok(format!(
-        r#""use strict";
-function __initialize() {{
-(function( factory ) {{
-    // stripped for screeps usage
-    factory();
-}}( function() {{
-    return (function( module_factory ) {{
-        // replaced with hardcoded grab for screeps usage
-        var instance = module_factory();
+        r#"{}
+
+/**
```

Comment:
Could the following also be extracted in their own template rather than hardcoded here?

Also, I'm not sure what I think of the comments in the js file, especially hardcoded ones. I know we are talking about a few bytes, but this is still bytes that could be used for code rather than for comments.

Reply:
Yeah, comments could definitely be removed. If we're that pressed for bytes, though, running a minimizer on main.js would remove these comments and more.

The reason this section is hardcoded right now is that it's so closely tied to `cargo-web` version. Particularly if the output format changes, we'll need to change both the prefix/suffix patterns and these functions to correctly get the right part of stdweb's wrapper.

We could pull it out into a template, but I don't see any need for customization, and we would have two or more places to modify when updating to new `cargo-web` changes. Then, if a user has customized it, they're customizations will break when updating `cargo-web` and `cargo-screeps`. I'm hoping by keeping this as part of `cargo-screeps` we can make it the more stable interface that further customizations can build on.